generated/nidcpower/nidcpower/lcr_load_compensation_spot.py

```diff
@@ -0,0 +1,118 @@
+        if self.reference_value_type == enums.LCRReferenceValueType.IMPEDANCE.value:
+            self.reference_value_a = data.reference_value.real
+            self.reference_value_b = data.reference_value.imag
+        else:
+            self.reference_value_a = data.reference_value
+            self.reference_value_b = 0.0
+
+
+class LCRLoadCompensationSpot(object):
+    """Constructs a DUT specification for a given frequency to use in LCR load compensation."""
+
+    def __init__(
+        self,
+        frequency=0.0,
+        impedance=None,
+        ideal_capacitance=None,
+        ideal_inductance=None,
+        ideal_resistance=None,
+    ):
+        """LCRLoadCompensationSpot
+
+        Creates and returns an LCRLoadCompensationSpot object. At most one of impedance,
+            ideal_capacitance, ideal_inductance and ideal_resistance can be set, and the remaining
+            parameters (excluding frequency) must be None. The parameter that is not None specifies
+            the known specification value of the DUT to be used as the basis for load compensation.
+            If all of them are None, then the default value of `impedance=complex()` will be used.
+
+        Args:
+            frequency (float): Specifies the spot frequency, in Hz.
+
+            impedance (complex): Specifies the actual impedance of your DUT to be used as the basis
+                for load compensation, or None to use another type of DUT specification value.
+
+            ideal_capacitance (float): Specifies the ideal capacitance of your DUT to be used as the
+                basis for load compensation, or None to use another type of DUT specification value.
+
+            ideal_inductance (float): Specifies the ideal inductance of your DUT to be used as the
+                basis for load compensation, or None to use another type of DUT specification value.
+
+            ideal_resistance (float): Specifies the ideal resistance of your DUT to be used as the
+                basis for load compensation, or None to use another type of DUT specification value.
+        """
+        self.frequency = frequency
+        # Set default values
+        self.reference_value_type = enums.LCRReferenceValueType.IMPEDANCE
+        self.reference_value = complex()
+        # Input validations
+        none_count = 0
+        for uppercase_parameter_name in enums.LCRReferenceValueType.__members__:
+            parameter_name = uppercase_parameter_name.lower()
+            parameter = locals()[parameter_name]
+            if parameter is None:
+                none_count += 1
+            elif parameter_name == "impedance" and not isinstance(parameter, numbers.Complex):
+                raise TypeError("Parameter impedance must be of type complex")
+            elif parameter_name != "impedance" and not isinstance(parameter, numbers.Real):
+                raise TypeError("Parameter {} must be a real number".format(parameter_name))
```

Comment:
This type checking is needed because the parameter would be assigned directly to `self.reference_value`, and the constructor of `struct_NILCRLoadCompensationSpot` assumes that `self.reference_value` is a complex number or a real number depending on `self.reference_value_type`. If we really want to support duck-typing in this case, I can remove the type validation and just let the constructor of `struct_NILCRLoadCompensationSpot` errors out when the user passes in the wrong data type while trying to call the `perform_lcr_load_compensation_spot()` method.

>  So what if the user passes impedance as integer and not numbers.Real??

Impedance need to be a complex number instead of a real number. Also, passing in an integer when a `numbers.Real` instance is expected would pass the type validation, since `numbers.Real` isn't a concrete type, and `isinstance()` checks for subtype relationship as well instead of an exact type match:
```
>>> isinstance(1, numbers.Real)
True
```

Reply:
I have removed the type validation in my latest commit.